source/App/libvvcdec/libvvcdec.h

```diff
@@ -0,0 +1,246 @@
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+ * Neither the name of Fraunhofer nor the names of its contributors may
+   be used to endorse or promote products derived from this software without
+   specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS
+BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+
+
+------------------------------------------------------------------------------------------- */
+
+#pragma once
+
+/** \file libvvcdec.h
+ * Decoder library public API definition.
+ * This public API defines the functions that are exported by the dynamic library libvvcdec.
+ * The API can be used like this:
+ * \code{.cpp}
+ * libvvcdec_context *decCtx = libvvcdec_new_decoder();
+ * bool eof = false;
+ * uint8_t *data = NULL;
+ * int length = -1;
+ * while (!eof)
+ * {
+ *   if (data == NULL && length == -1)
+ *   {
+ *     // Get the next NAL unit to push
+ *     data = get_nal_unit_data();
+ *     length = get_lenght_of_nal_unit_data();
+ *     eof = is_this_the_last_nal_unit();
+ *   }
+ *
+ *   bool check_output;
+ *   if (libvvcdec_push_nal_unit(decCtx, data, length, check_output) != LIBVVCDEC_OK)
+ *     handle_the_error();
+ *
+ *   // Next, check the output
+ *   if (check_output)
+ *   {
+ *     auto picData = libvvcdec_get_picture_plane(decCtx, LIBVVCDEC_LUMA);
+ *     if (!picData)
+ *       handle_error();
+ *     else
+ *     {
+ *       auto width = libvvcdec_get_picture_width(pic, LIBVVCDEC_LUMA);
+ *       auto height = libvvcdec_get_picture_height(pic, LIBVVCDEC_LUMA);
+ *       // ... Save the frame data
+ *     }
+ *   }
+ * }
+ * // Flush
+ * if (libvvcdec_push_nal_unit(decCtx, nullptr, 0, check_output) != LIBVVCDEC_OK)
+ *   handle_the_error();
+ * while(true)
+ * {
+ *   auto picData = libvvcdec_get_picture_plane(decCtx, LIBVVCDEC_LUMA);
+ *   if (!picData)
+ *     break;
+ *   // As above get the frame and do something with it.
+ *   if (libvvcdec_push_nal_unit(decCtx, nullptr, 0, check_output) != LIBVVCDEC_OK)
+ *     handle_the_error();
+ * }
+ * 
+ * libvvcdec_free_decoder(decCtx);
+ * \endcode
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+//#include <stdint.h>
+#include <cstdint>
```

Comment:
Any reason why this is using the C++ version of the header instead of the C if you have the `extern "C"` guard?

Reply:
Not 100% sure what the best practice here is. Found [this](https://stackoverflow.com/questions/13642827/cstdint-vs-stdint-h) where it says that the c version is deprecated and should not be used. Is there any advantage I am missing?

Reply:
I think it makes sense to use the C version here, because the header could be included by a C-only source file, in which case `cstdint` will not be available.
If `stdint.h` is deprecated in C++ most portable version would probably be to include on ore the other depending on if `__cplusplus` is defined.